plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -33,23 +43,47 @@ func (is *InternetSpeed) SampleConfig() string {
 
 const measurement = "internet_speed"
 
+var serverCache *speedtest.Server
+
 func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
-	user, err := speedtest.FetchUserInfo()
-	if err != nil {
-		return fmt.Errorf("fetching user info failed: %v", err)
-	}
-	serverList, err := speedtest.FetchServerList(user)
-	if err != nil {
-		return fmt.Errorf("fetching server list failed: %v", err)
+
+	// Sleep for offset duration
+	if is.Offset != "" {
+		d, err := time.ParseDuration(is.Offset)
+		if err != nil {
+			return fmt.Errorf("parsing offset failed: %v", err)
+		}
+		time.Sleep(d)
 	}
```

Comment:
With `is.Offset` being a `config.Duration` you can do
```suggestion
	time.Sleep(time.Duration(is.Offset))
```